File: benchmarks/word_count.go
package main

import (
	"fmt"
	"time"
)

func main() {
	t0 := time.Now()
	d := map[string]int{}
	for i := 0; i < 1000000; i++ {
		w := fmt.Sprintf("word%d", i%1000)
		d[w] = d[w] + 1
	}
	total := 0
	for _, v := range d {
		total += v
	}
	fmt.Println(total + len(d))
	fmt.Printf("bench_ns=%d\n", time.Since(t0).Nanoseconds())
}

File: benchmarks/word_count.js
const { performance } = require('perf_hooks');

const t0 = performance.now();
const d = new Map();
for (let i = 0; i < 1000000; i++) {
    const w = "word" + (i % 1000);
    d.set(w, (d.get(w) || 0) + 1);
}
let total = 0;
for (const v of d.values()) {
    total += v;
}
console.log(total + d.size);
console.log(`bench_ns=${Math.round((performance.now() - t0) * 1e6)}`);
